assets/app/scripts/services/ws.js

```diff
@@ -75,4 +75,16 @@ angular.module('openshiftConsole')
 
     return $ws;
   };
+})
+
+/* A WebSocket factory for kubernetesContainerTerminal */
+.factory("ContainerWebSocket", function(API_CFG, $ws) {
```

Comment:
@spadgett fyi with your auth changes... we might want this in data.js eventually